### ISSUE TITLE
Azure OpenAI Integration is not working anymore

### ISSUE THREAD

**Issue by niklasfink** [gpt-engineer-org/gpt-engineer#835]
The last working version I think was around v0.0.6. v0.1.0 is not working anymore with the following effects:

## Expected Behavior

When using the `--azure` parameter, the Azure OpenAI endpoint should be used. (X.openai.azure.com)

## Current Behavior

Instead of the Azure OpenAI endpoint, the usual OpenAI endpoint is being used. (api.openai.com)

## Failure Information

```
gpt-engineer --azure https://X.openai.azure.com -v ./projects/example/ gpt-4-32k
DEBUG:openai:message='Request to OpenAI API' method=get path=https://api.openai.com/v1/models/gpt-4-32k
DEBUG:openai:api_version=None data=None message='Post details'
DEBUG:urllib3.util.retry:Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.openai.com:443
DEBUG:urllib3.connectionpool:https://api.openai.com:443 "GET /v1/models/gpt-4-32k HTTP/1.1" 401 262
DEBUG:openai:message='OpenAI API response' path=https://api.openai.com/v1/models/gpt-4-32k processing_ms=3 request_id=XX response_code=401
INFO:openai:error_code=invalid_api_key error_message='Incorrect API key provided: XX. You can find your API key at https://platform.openai.com/account/api-keys.' error_param=None error_type=invalid_request_error message='OpenAI API error received' stream_error=False
```

### Steps to Reproduce

1. Use the command as seen above with the --azure parameter.
2. Use verbose output to verify that it's not going to the AzOAI Endpoint, but to the default OAI endpoint.

**Comment by niklasfink** [comment]
Issue is fixed when changing the following code
https://github.com/AntonOsika/gpt-engineer/blob/1408652958b7a77b49990dc02df3450656c680d7/gpt_engineer/core/ai.py#L105

into this:
```
self.model_name = (
    model_name
    if azure_endpoint
    else self._check_model_access_and_fallback(model_name)
)
```

**Comment by ATheorell** [comment]
If you make a PR of this, I will merge it.

**Comment by captivus** [comment]
Prompting @niklasfink to submit a PR. We're keen for contributors!

**Comment by ATheorell** [comment]
@niklasfink I guess this is still an open thing? I don't have access to an azure instance so I cannot really test this. Would be really cool if you could make  working PR.

**Comment by niklasfink** [comment]
Sorry, I don’t have the capacity at the moment to test and create a proper PR. I’ll try to do so in January.

**Comment by ATheorell** [comment]
> Sorry, I don’t have the capacity at the moment to test and create a proper PR. I’ll try to do so in January.

Fair, in case no one else does it before that, a PR in January is most welcome

**Comment by praveen-palanisamy** [comment]
This issues was already fixed in 46643f62f45948478795599b476b0781f3d35ebf and since [v0.2.5](https://github.com/gpt-engineer-org/gpt-engineer/releases/tag/v.0.2.5) the Azure OpenAI integration is working as expected.
That commit removed the `_check_model_access_and_fallback` method altogether. Here's the relevant diff:
![image](https://github.com/gpt-engineer-org/gpt-engineer/assets/4770482/ac98a489-b36c-4861-a156-11de664a70ba)

The following is tested to work:
`gpt-engineer --azure https://YOUR_DEPLOYED_ENDPOINT_ID.openai.azure.com -v ./projects/example/ gpt-4-32k`

```
DEBUG:gpt_engineer.core.ai:Using model gpt-4-32k
DEBUG:gpt_engineer.core.ai:Creating a new chat completion: [SystemMessage(content='...'), HumanMessage(content='...')]
DEBUG:openai:message='Request to OpenAI API' method=post path=https://YOUR_DEPLOYED_ENDPOINT_ID.openai.azure.com/openai/deployments/gpt-4-32k/chat/completions?api-version=YYYY-MM-DD
DEBUG:openai:api_version=YYYY-MM-DD
```

@niklasfink : Please use/test-with [v0.2.5](https://github.com/gpt-engineer-org/gpt-engineer/releases/tag/v.0.2.5) (or later versions).

@ATheorell : The `bug` label can be removed. The issue can be closed (after feedback from @niklasfink ).

**Comment by ATheorell** [comment]
@praveen-palanisamy thank you for the convincing and well established feedback! Closing!
